cde/sdk/go/pkg/cdf/events/event_types.go

```diff
@@ -0,0 +1,269 @@
+
+	event.SetData(cloudevents.ApplicationJSON, artifactData)
+
+	return event, nil
+}
+
+func setExtensionForBuildEvents(event cloudevents.Event, buildId string, buildName string, buildArtifactId string) {
+	event.SetExtension("buildid", buildId)
+	event.SetExtension("buildname", buildName)
+	event.SetExtension("buildartifactid", buildArtifactId)
+
+}
+
+func CreateBranchEvent(eventType CDEventType,
+	branchId string,
+	branchName string,
+	branchRepositoryId string,
+	branchData map[string]string) (cloudevents.Event, error) {
+	event := cloudevents.NewEvent()
+	event.SetID(uuid.NewV4().String())
+	event.SetType(eventType.String())
+	event.SetTime(time.Now())
+
+	setExtensionForBranchEvents(event, branchId, branchName, branchRepositoryId )
+
+	event.SetData(cloudevents.ApplicationJSON, branchData)
+
+	return event, nil
+}
+
+func setExtensionForBranchEvents(event cloudevents.Event, branchId string, branchName string, branchRepositoryId string) {
+	event.SetExtension("branchid", branchId)
+	event.SetExtension("branchname", branchName)
+	event.SetExtension("branchrepositoryid", branchRepositoryId)
+
+}
+
+func CreateRepositoryEvent(eventType CDEventType,
+	repositoryId string,
+	repositoryName string,
+	repositoryURL string,
+	repositoryData map[string]string) (cloudevents.Event, error) {
+	event := cloudevents.NewEvent()
+	event.SetID(uuid.NewV4().String())
+	event.SetType(eventType.String())
+	event.SetTime(time.Now())
+
+	setExtensionForRepositoryEvents(event, repositoryId, repositoryName, repositoryURL )
+
+	event.SetData(cloudevents.ApplicationJSON, repositoryData)
+
+	return event, nil
+}
+
+func setExtensionForRepositoryEvents(event cloudevents.Event, repositoryId string, repositoryName string, repositoryURL string) {
+	event.SetExtension("repositoryid", repositoryId)
+	event.SetExtension("repositoryname", repositoryName)
+	event.SetExtension("repositoryurl", repositoryURL)
+
+}
+
+func CreateTaskRunEvent(eventType CDEventType,
+	taskRunId string,
+	taskRunName string,
+	taskRunPipelineId string,
+	taskRunData map[string]string) (cloudevents.Event, error) {
+	event := cloudevents.NewEvent()
+	event.SetID(uuid.NewV4().String())
+	event.SetType(eventType.String())
+	event.SetTime(time.Now())
+
+	setExtensionForTaskRunEvents(event, taskRunId, taskRunName, taskRunPipelineId )
+
+	event.SetData(cloudevents.ApplicationJSON, taskRunData)
+
+	return event, nil
+}
+
+func setExtensionForTaskRunEvents(event cloudevents.Event, taskRunId string, taskRunName string, taskRunPipelineId string) {
+	event.SetExtension("taskrunid", taskRunId)
+	event.SetExtension("taskrunname", taskRunName)
+	event.SetExtension("taskrunpipelineid", taskRunPipelineId)
+
+}
+
+func CreatePipelineRunEvent(eventType CDEventType,
+	pipelineRunId string,
+	pipelineRunName string,
+	pipelineRunStatus string,
+	pipelineRunURL string,
+	pipelineRunErrors string,
+	pipelineRunData map[string]string) (cloudevents.Event, error) {
+	event := cloudevents.NewEvent()
```

Comment:
NIT: with this formatting I find it really hard to separate the end of the `func` signature from its body at a glance.

Reply:
I totally get what you mean here.. not sure what are Go best practices around that. Just for others to see it:
```
func CreateArtifactEvent(eventType CDEventType,
	artifactId string,
	artifactName string,
	artifactVersion string,
	artifactData map[string]string) (cloudevents.Event, error) {
```

vs maybe something like: 
```
func CreateArtifactEvent(eventType CDEventType, params ArtifactEventParams) (cloudevents.Event, error) {
```

Reply:
Yeah, that sounds good, that way we can keep the same interface even if we add/remove parameters in future